core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -151,11 +164,49 @@ oppia.directive('searchBar', [function() {
             }
 
             _updateSelectionDetails('languageCodes');
+
+            if (Boolean(urlService.getUrlParams().q)) {
+              _parseURL();
+            }
             _onSearchQueryChangeExec();
 
             _searchBarFullyLoaded = true;
           }
         );
+
+        var _parseURL = function() {
+          var urlSearch = decodeURI($window.location.search);
+          var query = urlSearch.replace('?q=', '');
+
+          // Grab language code(s) from URL, these are 2 lowercase letters.
+          var _languageCodeSelections = query.match(/"\w\w"/g);
```

Comment:
I would expect to see language_code somewhere in the regex (or the logic for extracting the language codes out). I mean, what if the query that's typed in contains something like "ab"?